src/state/user/hooks.tsx

```diff
@@ -68,6 +69,18 @@ export function useIsExpertMode(): boolean {
   return useAppSelector((state) => state.user.userExpertMode)
 }
 
+export function useBuyFiatClicked(): [boolean | undefined, (buyFiatClicked: boolean) => void] {
+  const dispatch = useAppDispatch()
+  const buyFiatClicked = useAppSelector((state) => state.user.buyFiatClicked)
+  const setBuyFiatClicked = useCallback(
+    (buyFiatClicked: boolean) => {
+      dispatch(updateUserBuyFiatClicked({ buyFiatClicked }))
```

Comment:
you can just pass in the `boolean` value as the payload here right?

```
dispatch(updateUserBuyFiatClicked(buyFiatClicked))`
```

and then the reducer below would just be

```
updateUserBuyFiatClicked(state, action) {
      state.buyFiatClicked = action.payload
    },
```